src/Avalonia.Base/AvaloniaProperty.cs

```diff
@@ -36,27 +38,22 @@ public abstract class AvaloniaProperty : IEquatable<AvaloniaProperty>, IProperty
             Type valueType,
             Type ownerType,
             AvaloniaPropertyMetadata metadata,
-            Action<IAvaloniaObject, bool> notifying = null)
+            Action<IAvaloniaObject, bool>? notifying = null)
         {
-            Contract.Requires<ArgumentNullException>(name != null);
-            Contract.Requires<ArgumentNullException>(valueType != null);
-            Contract.Requires<ArgumentNullException>(ownerType != null);
-            Contract.Requires<ArgumentNullException>(metadata != null);
-
             if (name.Contains("."))
```

Comment:
You use "name" here before null check below.

Reply:
Good catch. Fixed.